processing/pom.xml

```diff
@@ -34,6 +34,11 @@
   </properties>
 
   <dependencies>
+    <dependency>
+      <groupId>org.apache.spark</groupId>
+      <artifactId>spark-core_${scala.binary.version}</artifactId>
+      <version>${spark.version}</version>
+    </dependency>
```

Comment:
Remove this dependency. Do not add spark dependency in processing layer until and unless really required